dask/array/ufunc.py

```diff
@@ -29,7 +33,12 @@ def wrap_elemwise(numpy_ufunc, array_wrap=False):
     def wrapped(*args, **kwargs):
         dsk = [arg for arg in args if hasattr(arg, '_elemwise')]
         if len(dsk) > 0:
-            if array_wrap:
+            try:
+                is_dataframe = isinstance(dsk[0], ddcore._Frame)
+            except NameError:
+                is_dataframe = False
```

Comment:
Maybe we should change this to use `is_dataframe_like` and/or `is_series_like ` on `x`? Perhaps that resolves the dataframe issue.

Reply:
@jakirkham thanks for the suggestion. It seems a good idea to use is_dataframe_like(dsk[0]) instead of isinstance(dsk[0], ddcore._Frame). But we still need to keep the try/except block (in case we couldn't import dask.dataframe.utils.is_dataframe_like). Am I missing a simpler alternative, or is that what you suggested?

Reply:
What if we import `is_dataframe_like` directly and then add a shim function when it is not defined? Would that address this issue or is there something else I'm missing?

Reply:
is_dataframe_like is often used to type check things like pandas/cudf dataframes.  It's also possible to use it to type check for dask dataframe objects, but often not necessary because we're able to import those types directly.  I don't immediately see the connection between `is_dataframe_like` and the issues raised above, but I'm also not at all familiar with the logic behind `__array_wrap__`.

Reply:
The goal of this change is to ensure `__array_wrap__` is always used when dataframes are involved. This should allow Pandas to determine the behavior without us influencing it. My guess is this fixes the issue Peter identified.

Reply:
@jakirkham is correct. Since `__array_function__` is not implemented in Pandas yet, we want to ensure it still uses `__array_wrap__`. This allows us to move Dask to `__array_function__` without requiring Pandas to implement that for the time being.

Reply:
Grand.  I'll just sneak away from this PR then :)

Reply:
When I say Dask, I mean strictly `dask.Array`. `dask.dataframe` would continue to use `__array_wrap__`, since it uses Pandas by default.

Reply:
> What if we import `is_dataframe_like` directly and then add a shim function when it is not defined? Would that address this issue or is there something else I'm missing?

I think it would solve it, yes. This function would have to live somewhere in `dask/array`. To make sure I understand your suggestion, would it be something like the following?

```python
def is_dataframe_like(df):
    try:
        from ..dataframe.utils import is_dataframe_like
        return is_dataframe_like(df)
    except ImportError:
        return false
```

If so, I would suggesting to have it either in `dask/array/ufunc.py` itself, or `dask/array/utils.py`. Thoughts?

Reply:
That's the general idea, I'd probably tweak it to look like this...

```python
try:
    from ..dataframe.utils import is_dataframe_like
except ImportError:
    def is_dataframe_like(df):
        return False
```

Reply:
I'm ok defining this at a higher level as well if you think it would help. Am also ok leaving it here for now.

Reply:
@jakirkham I see now. This indeed feels cleaner, thanks for clarifying!